pkg/api/types.go

```diff
@@ -1428,6 +1430,26 @@ func (cs *ContainerService) GetLocations() []string {
 	return allLocations
 }
 
+// GetAuthenticationMethod returns authentication method which k8s azure cloud provider will use
```

Comment:
Is this function ever used in a non-Azure Stack context? If not, it would make more sense to name it something like `GetCustomCloudAuthenticationMethod `.